src/lmic/radio.c

```diff
@@ -25,6 +25,10 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#if LMIC_X_DEBUG_LEVEL > 0
```

Comment:
This change breaks non-Arduino uses of this library. I think i will have to back it out. I imagine this is needed because you're using a different BSP than I am. Can you let me know what the issue is and we can do this in the `.h` files somehow?

Reply:
The reason why I introduce LMIC_X_DEBUG_LEVEL is to reduce the size of SRAM.  When I enabled LMIC_DEBUG_LEVEL, the total size was too big to run my application code. 
 Therefore, I have added my own debugging code.  I haven't used the original debug code of LMIC_DEBUG_LEVEL at all.  Some of code might be merged into one.

Reply:
That's what I guessed. I tried to merge based on this assumption. Are you using an AVR CPU? I guess that the printf-formats need to be in flash for AVR, which means using a slightly different print routine. I'm using mostly ARM, but don't want to break the AVR use cases unnecessarily.

Reply:
Yes, it's Dragino Mini Dev (ATmega328P).  It didn't help much when I disabled the class B staff actually.  Thanks for preserving the AVR use cases.